matchzoo/data_pack/data_pack.py

```diff
@@ -294,6 +294,34 @@ def drop_label(self):
         """
         self._relation = self._relation.drop(columns='label')
 
+    @_optional_inplace
+    def drop_invalid(self):
+        """
+        Remove rows from the data pack where the length is zero.
+
+        :param inplace: `True` to modify inplace, `False` to return a modified
+            copy. (default: `False`)
+
+        Example:
+            >>> import matchzoo as mz
+            >>> data_pack = mz.datasets.toy.load_data()
+            >>> data_pack.append_text_length(inplace=True, verbose=0)
+            >>> data_pack.drop_invalid(inplace=True)
+        """
+        if not ('length_left' in self._left and 'length_right' in self._right):
+            raise ValueError(f"`length_left` or `length_right` is missed. "
+                             f"Please call `append_text_length` in advance.")
+        valid_left = self._left.loc[
+            self._left.length_left != 0].reset_index()
+        valid_right = self._right.loc[
+            self._right.length_right != 0].reset_index()
+        self._left = self._left[self._left.index.isin(valid_left.id_left)]
```

Comment:
do you think it's correct? the `id_left` corresponds to the id in the original `left` dataframe, however, the `id` has been `reset_index` in line 314.

Reply:
The `id` in `left` is not equal to the `id` in `valid_left`.

Reply:
Yeah.  Since I have `reset_index` in line 315 and 316. Thus, I can use `isin(valid.id_left)`. I have tested these lines on the toy data. Maybe these codes can be simplified by drop `reset_index()`, then use `isin(valid_left.id)`?